torch/csrc/distributed/c10d/FlightRecorder.cpp

```diff
@@ -678,6 +670,58 @@ std::string FlightRecorder<EventType>::dump(
     result.insert(
         entries_key, getCollectiveTrace(includeStackTraces, onlyActive));
   }
+  return result;
+}
+
+template <typename EventType>
+std::string FlightRecorder<EventType>::dump_json(
+    bool includeCollectives,
+    bool onlyActive) {
+  return (get_dump_json(includeCollectives, onlyActive)).dump();
+}
+
+template <typename EventType>
+// dump all collectives + ncclDumpMap
+std::string FlightRecorder<EventType>::dump(
+    bool includeCollectives,
+    bool includeStackTraces,
+    bool onlyActive) {
+  return pickle_str(
+      get_dump(includeCollectives, includeStackTraces, onlyActive));
+}
+
+std::unique_ptr<DebugInfoWriter> DebugInfoWriter::writer_ = nullptr;
+std::atomic<bool> DebugInfoWriter::hasWriterRegistered_(false);
+
+float getDurationFromEvent(c10::Event& startEvent, c10::Event& endEvent) {
+  TORCH_CHECK(false, "getDuration not supported by c10::Event.");
+}
```

Comment:
That's a surprise. cc @albanD

Reply:
now you see my point... I try the way cuda event is doing but I got compile error.